modules/apps/headless/headless-asset-library/headless-asset-library-impl/src/main/java/com/liferay/headless/asset/library/internal/resource/v1_0/UserAccountResourceImpl.java

```diff
@@ -130,9 +139,11 @@ public UserAccount getAssetLibraryUserAccount(
 			throw new UnsupportedOperationException();
 		}
 
-		User user = _userService.getUserById(userAccountId);
+		_checkAssetLibraryUser(assetLibraryId);
 
-		if (!_groupService.hasUserGroup(user.getUserId(), assetLibraryId)) {
+		User user = _userLocalService.getUserById(userAccountId);
+
+		if (!_userLocalService.hasGroupUser(assetLibraryId, user.getUserId())) {
 			throw new NoSuchUserException(
 				StringBundler.concat(
 					"User ", userAccountId, " is not associated to group ",
```

Comment:
Can you extract this into a `_check*` method? You can reuse this logic inside the other _check method.

Reply:
Done!